src/libs/SearchUtils.ts

```diff
@@ -301,7 +304,31 @@ function isSearchResultsEmpty(searchResults: SearchResults) {
     return !Object.keys(searchResults?.data).some((key) => key.startsWith(ONYXKEYS.COLLECTION.TRANSACTION));
 }
 
+// This function should replace the current getQueryParams after the search v2.1 implementation.
+function getQueryHashFromString(query: string): number {
+    return UserUtils.hashText(query, 2 ** 32);
+}
+
+type JSONQuery = {
+    input: string;
+    hash: number;
+};
+
+function buildJSONQuery(query: string) {
+    try {
+        // Add the full input and hash to the results
+        const result = searchParser.parse(query) as JSONQuery;
+        result.input = query;
+        result.hash = getQueryHashFromString(query);
+        return result;
+    } catch (e) {
+        // TODO: We need better error handling here.
```

Comment:
Let's remove this too. I'll create a follow up issue for it

Reply:
done